v8/config/krb5conf.go

```diff
@@ -507,7 +508,11 @@ func (c *Config) ResolveRealm(domainName string) string {
 			return r
 		}
 	}
-	return c.LibDefaults.DefaultRealm
+
+	if useDefault {
+		return c.LibDefaults.DefaultRealm
+	}
+	return ""
```

Comment:
I'm wondering is the simpler fix is to just have this function return a blank string all the time if nothing if matched. I think the only time this is used is when performing a TGS exchange.